nexus/internal/uploader/upload_task.go

```diff
@@ -0,0 +1,33 @@
+package uploader
+
+import "sync"
+
+// UploadTask is a task to upload a file
+type UploadTask struct {
+
+	// Path is the path to the file
+	Path string
+
+	// Url is the endpoint to upload to
+	Url string
+
+	// Headers to send on the upload
+	Headers []string
+
+	// allow tasks to wait for completion (failed or success)
+	WgOutstanding *sync.WaitGroup
+}
+
+func (t *UploadTask) outstandingAdd() {
+	if t.WgOutstanding == nil {
```

Comment:
question to think about... should this be optional?  Do we ever want tasks to not wait for completion?